src/openrct2/ride/Vehicle.cpp

```diff
@@ -7273,6 +7275,16 @@ void Vehicle::UpdateAnimationAnimalFlying()
     }
 }
 
+uint8_t Vehicle::GetTargetFrame(rct_ride_entry_vehicle* vehicleEntry)
+{
+    animationState += _vehicleVelocityF64E08;
+    // Similar to a modulus, but using a mask and multiplying before a final bit shift to get rid of the mask bits. You are left
+    // with a value from 0 to AnimationFrames - 1. Prevents skipped frames on non powers-of-two AnimationFrames.
+    return (((animationState >> std::max(vehicleEntry->AnimationSpeed - UINT8_WIDTH, 0)) & UINT8_MAX)
```

Comment:
Don't use C99, std::numeric_limits<T> is the C++ way.

Reply:
this comment is greek to me

Reply:
What member function do I call to get width? `std::numeric_limits<uint8_t>::width()`?

Reply:
Alright, you got your wish.